src/kernel_abi.h

```diff
@@ -497,9 +497,18 @@ struct BaseArch : public wordsize, public FcntlConstants {
   struct shmid64_ds {
     ipc64_perm shm_perm;
     size_t shm_segsz;
-    uint64_t shm_atime_only_little_endian;
-    uint64_t shm_dtime_only_little_endian;
-    uint64_t shm_ctime_only_little_endian;
+    __kernel_time_t shm_atime;
+#if __BITS_PER_LONG != 64
+    unsigned long unused1;
+#endif
```

Comment:
This doesn't look right. `__BITS_PER_LONG` here would be rr's `__BITS_PER_LONG`; to emulate what `asm-generic/shmbuf.h` is doing, we'd need to make this conditional on the `wordsize` template parameter, which I don't think we can do easily.

I think the simplest thing to do is just provide separate definitions per architecture.